pkgs/by-name/ma/mautrix-gmessages/package.nix

```
@@ -0,0 +1,38 @@
+{
+  fetchFromGitHub,
+  lib,
+  buildGoModule,
+  olm,
+  # This option enables the use of an experimental pure-Go implementation of
+  # the Olm protocol instead of libolm for end-to-end encryption. Using goolm
+  # is not recommended by the mautrix developers, but they are interested in
+  # people trying it out in non-production-critical environments and reporting
+  # any issues they run into.
+  withGoolm ? false,
+}:
+
+buildGoModule rec {
+  pname = "mautrix-gmessages";
+  version = "0.5.2";
+  src = fetchFromGitHub {
+    owner = "mautrix";
+    repo = "gmessages";
+    rev = "v${version}";
+    hash = "sha256-3sx2XxzH/cj2zJIkV9QKKVadxSWwf4YZbdg3p4repqs=";
+  };
+
+  vendorHash = "sha256-8KexkPsRqzlVeYMF0dVgDMS38zQ3i+erQgsdgHDOCF0=";
+
+  buildInputs = lib.optional (!withGoolm) olm;
+  tags = lib.optional withGoolm "goolm";
+
+  doCheck = false;
```

Indicate reason like

```suggestion
  doCheck = false; # Require network access
```

---

Actually the unit tests of this package works correctly. I copied this line from `mautrix-whatsapp` and didn't test the unit test at all.